src/components/ProductItem/ProductItem.ts

```diff
@@ -0,0 +1,121 @@
+
+export let cart: string[] = [];
+const productCounter = document.querySelector(".productCounter");
+const headerLink = document.querySelector(".header__link");
+
+const total = document.querySelector(".total");
+
+let amount = 0;
+
+export class ProductItem implements appComponents {
+  constructor(private product: Product) {}
+
+  getId = () => `${this.product.id}`;
+  getIds = () => `aaa`;
+
+  render() {
+    return `
+      <div class="card" style="width: 18rem; margin: 1em;">
+        <img src="${this.product.image}" alt="${
+      this.product.name
+    }" class="card-img-top" style="backgroung-size: auto;"/>
+        <div class="card-body">
+          <h5 class="card-title">${this.product.name}</h5>
+          <p class="card-text">${this.product.price}</p>
+            <a href="#" class="btn_add btn btn-primary" id=${this.getId()}>BUY</a>
+        </div>
+      </div>
+    `;
+  }
+
+  addEvents() {
+    const buttonItem = document.getElementById(this.getId())!;
+    if (buttonItem == null && location.hash !== "#cart") {
+      throw new Error("!!!");
+    } else if (location.hash !== "#cart") {
+      buttonItem.addEventListener("click", () => {
+        if (buttonItem.classList.contains("btn_delete")) {
+          buttonItem.classList.remove("btn_delete");
+          buttonItem!.innerHTML = "BUY";
+          cart = cart.filter((product) => !product.match(this.product.image));
+          productCounter!.innerHTML = String(cart.length);
+          amount -= this.product.price;
+          total!.innerHTML = String(amount);
+        } else {
+          buttonItem.classList.add("btn_delete");
+          buttonItem!.innerHTML = "DROP FROM CART";
+          cart.push(`<div class="${this.product.id}">
+            <div class="">
+              <div class="col-md-4">
+                <img src="${
+                  this.product.image
+                }" class="img-fluid rounded-start" style="max-width: 10rem;" alt="${
+            this.product.name
+          }">
+              </div>
+              <div class="col-md-8">
+                <div class="card-body">
+                  <h5 class="card-title">${this.product.name}</h5>
+                  <h6 class="card-title">$${this.product.price}</h6>
+                  <h6 class="card-title">${this.product.stock}</h6>
+                  <div class='stock'>
+                  <button class='btn btn-dark' id='${this.getIds()}' onclick="
+                  const stockCounter = document.getElementById('${
+                    "stockCounter" + this.getId()
+                  }')
+                  const total = document.querySelector('.total')
+                  const productCounter = document.querySelector('.productCounter')
+                  if(stockCounter.innerHTML < ${this.product.stock}){
+                   stockCounter.innerHTML++
+                   total.innerHTML= total.innerHTML -0 + ${this.product.price}
+                   productCounter.innerHTML++
+                  }
```

Comment:
What about `addEventListener`? It's really hard to understand what's going on here